packages/xrpl/src/models/utils/flags.ts

```diff
@@ -60,34 +58,50 @@ const txToFlag = {
 /**
  * Sets a transaction's flags to its numeric representation.
  *
+ * @deprecated
+ * This utility function is deprecated.
+ * Use convertTxFlagsToNumber() instead and use the returned value to modify the tx.Flags from the caller.
+ *
  * @param tx - A transaction to set its flags to its numeric representation.
  */
 export function setTransactionFlagsToNumber(tx: Transaction): void {
-  if (tx.Flags == null) {
-    tx.Flags = 0
-    return
-  }
+  // eslint-disable-next-line no-console -- intended deprecation warning
+  console.warn(
+    'This function is deprecated. Use convertTxFlagsToNumber() instead and use the returned value to modify the tx.Flags from the caller.',
+  )
+
+  // eslint-disable-next-line no-param-reassign -- intended param reassign in setter
+  tx.Flags = convertTxFlagsToNumber(tx)
+}
+
+/**
+ * Returns a transaction's flags as its numeric representation.
+ *
+ * @param tx - A transaction to parse flags for
+ * @returns A numerical representation of a transaction's flags
+ */
+export function convertTxFlagsToNumber(tx: Transaction): number {
   if (typeof tx.Flags === 'number') {
```

Comment:
missing a `null` check

Reply:
There is a check for nulls on line 91 before the conversion logic, do you mean a null check for the tx param as a whole?

Reply:
I believe she's referring to the updated behavior of the `setTransactionFlagsToNumber` function.

In the existing code, if `tx.Flags != null`, then the `tx` object is not updated at all.
However, in your PR, the `tx.Flags` will be over-written with the output of the `convertTxFlagsToNumber`

Reply:
I was under the impression that the numerical flag value 0 is equivalent to tx.Flags = null, which is what would be the result of the current code changes. Is representing a null tx.Flags value as null numerically what we would want instead? In that case I'll add an error as well since I don't think we would ever want that

Reply:
Shouldn't the last assert statement pass in this example?
```
tx.Flags = null
setTransactionFlagsToNumber(tx)
assert(tx.Flags === 0) // false
```

Reply:
Yes, and it still should since line 98 `if (tx.flags) ` wouldn't be true, so `convertTxFlagsToNumber` would return 0. In the old setter logic , if tx.Flags was null, it returned 0 as well so the behavior is the same, just without an explicit null check and early return